src/map/battle.c

```diff
@@ -5106,15 +5108,18 @@ static struct Damage battle_calc_weapon_attack(struct block_list *src, struct bl
 				break;
 		}
 
-		if( sd ) {
+		if (sd) {
 			// Weaponry Research hidden bonus
-			if ((temp = pc->checkskill(sd,BS_WEAPONRESEARCH)) > 0)
-				hitrate += hitrate * ( 2 * temp ) / 100;
+			if ((temp = pc->checkskill(sd, BS_WEAPONRESEARCH)) > 0)
+				hitpercbonus += 2 * temp;
 
 			if ((sd->weapontype == W_1HSWORD || sd->weapontype == W_DAGGER) && (temp = pc->checkskill(sd, GN_TRAINING_SWORD)) > 0)
 				hitrate += 3 * temp;
 		}
 
+		if (hitpercbonus > 0)
```

Comment:
I think it's fine if it's negative too. May have seen negative hitpercbonus somewhere in Aegis.

Reply:
Currently there's no way for hitpercbonus to be negative and including negative values might be confusing for whoever reads it in the future. I'd rather keep it like that until a case that forces it to be negative is included.